Puppi isolation electrons and photons miniAOD rebased

ishvetso opened the pull request:
This is a rebase of #15528
most of things are fixed, expect that as far as I understand patPhoton -> associatedPackedPFCandidates() is giving an error

mariadalfonso committed:
implement m2/m3 with HBHEChannelInfo phase1

mariadalfonso committed:
float to double from slava, 20% speed up on ttbar

mariadalfonso committed:
remove unused parameters (from Stephanie)

mariadalfonso committed:
remove hard coded parameters since other options never got tested(from Stephanie)

mariadalfonso committed:
remove the unused/untested pedelstal algos for M3

mariadalfonso committed:
add the first version the SiPM shape in the M2

smorovic committed:
Per-pid CMSSW stopping mechanism

smorovic committed:
print more detailed error on failed to parse JSON file

mariadalfonso committed:
use the new SiPM pulseShape

kpedro88 committed:
customize digi collections for hcal validation & dqm

kpedro88 committed:
generate emap for any topology from raw DetId bits when using hardcode conditions

mariadalfonso committed:
adjust the AC granularity and 1-3 pulse switch for siPM

kpedro88 committed:
remove old hardcoded emap, add ZDC and TrigTower to auto emap

kpedro88 committed:
alternate method for automatic emap

mariadalfonso committed:
fix missing comma

mariadalfonso committed:
add M0-Fix from Salavat

mariadalfonso committed:
handle the M2 parameters differently for HPD and siPM

mariadalfonso committed:
use heap memory saver solution instead of vector

mariadalfonso committed:
cleanup

mariadalfonso committed:
add placeholder for ADCnoise for QIE10

mariadalfonso committed:
removed unused parameters also in run2 code

mariadalfonso committed:
reorganize M2 Pset common to Run2/Phase1

mariadalfonso committed:
add assert to avoid division to zeroes

mariadalfonso committed:
fix the PR to restore correct timing info inside the recHit

ishvetso committed:
add CITK producers for electrons

ishvetso committed:
add module for electrons

ishvetso committed:
fix name

ishvetso committed:
update for PUPPI isolation

ishvetso committed:
fix naming

mariadalfonso committed:
M3-HLT: added protection for potential divisions by zero

ishvetso committed:
update for puppi isolation for electrons+renaming

lgray committed:
Allow CMS Phase2 with HGCal to run without crashing

makortel committed:
Enable pixel digi2raw and raw2digi modules in 2017 workflows

makortel committed:
Add matrix workflow for Run2_2016_HIPM era

makortel committed:
Customize CCC with era

makortel committed:
Customize MinPtForHitRecoveryInGluedDet with era

awhitbeck committed:
removing commented out code

mariadalfonso committed:
fix the HO-M0 time

mariadalfonso committed:
fix division for zeros

mariadalfonso committed:
set the pulseShapeID and pedestal Variation inside the HBHEChannelInfo

kirschen committed:
add precaution for deactivated HO (theHOIsUsed==false): set tower energy to tiny value to avoid nan-eta for HO-only towers (when HO is'deactivated', i.e. energy set to zero)

mariadalfonso committed:
store M2 recHit energy by default for Phase1

kirschen committed:
Revert "add precaution for deactivated HO (theHOIsUsed==false): set tower energy to tiny value to avoid nan-eta for HO-only towers (when HO is'deactivated', i.e. energy set to zero)"

This reverts commit 026a4e18f13086661453b0cf95244296733af2f0.

kirschen committed:
do check for unlikely HO-only tower, when putting the tower into the collection, following Slava's suggestion

uhussain committed:
Fixed logic error in line 1154 of SiStripElectronAlgo.cc - A null C++ object pointer was being called

uhussain committed:
Fixed logic error in line 208 in InOutConversionSeedFinder.cc - A null C++ object pointer was being called

lgray committed:
working on integrating sim PF and barrel PF together

ishvetso committed:
commit for review

ishvetso committed:
rename

ishvetso committed:
remove cout

gpetruc committed:
Get reducedEgamma working with no ES rechits in Phase2

kpedro88 committed:
set trigger bit in auto emap

kpedro88 committed:
fix unnecessary ptr copy in emap

ferencek committed:
more appropriate definition of loadWithPrefix and loadWithPostfix functions

roger-wolf committed:
switch bend corrections off for all 2prong and 3prong decays -- for decay modes with 0pi0s the switch has no effect. It is made here nevertheless for the sake of it.

roger-wolf committed:
small fix to make the high-level run in scheduled mode possible

roger-wolf committed:
cleaning up commented lines

roger-wolf committed:
and one more minor change to get things nice -- sorry

rappoccio committed:
Forgot one L1 correction for PUPPI. Also fixing validation + dqm sequences to look at new collections.

kirschen committed:
change behavior only for theHOIsUsed==false to keep backwards-compatible; N.B.: I assume towerP4[3]==0 && E_outer>0 should only happen, if theHOIsUsed==false anyway

Dr15Jones committed:
Unit test for serializing modules

Unit test checks that no modules are run concurrently in a job where we only have legacy and one modules using shared resources.

ishvetso committed:
minor

lgray committed:
minor reorganization to simplify propagation of merged PF collections

lgray committed:
update HGCal workflows to global reco

ishvetso committed:
few fixes

lgray committed:
get matrix tests running again

awhitbeck committed:
adding reference to FEDRawData

ishvetso committed:
fix fillDescription

ferencek committed:
relaxing the default b-tagging track selection

gpetruc committed:
Add strip and pixel layers to the miniAOD packed candidates

ferencek committed:
with relaxed selection as new default, making the legacy tight selection available in PAT

ferencek committed:
propagating latest b-tag updates to MiniAOD

gpetruc committed:
Monitoring of layers in PackedCandidateTrackValidator

gpetruc committed:
PackedCandidateTrackValidator: Include monitoring plots of numbers of layers from HitPattern

ishvetso committed:
logic fixed

ebrondol committed:
Merged phase2trackingWithCPE_InputTag_81X from repository ebrondol

gpetruc committed:
PackedCandidate::bestTrack with the right number of layers

ishvetso committed:
fill description is added

ishvetso committed:
PUPPI iso disregarded

ebrondol committed:
using eras

gpetruc committed:
Fix check on bestTrack(). packedLayers_ is not going to be zero for new miniAOD, but will be zero when reading old ones, so better check both that and packedHits_ (which may be zero on new miniAODs if there's just one hit per layer)

Commit:
clean up

Commit:
clean up extra line

lgray committed:
track necessary files...

Dr15Jones committed:
Improvements to SerialTaskQueueChain

-added move based functions
-pushAndWait will rethrow an exception thrown in the lambda
-added numberOfQueues() to be used by SharedResourcesAcquirer

Dr15Jones committed:
Use SerialTaskQueueChains for serialization

Switched from using std::mutex to SerialTaskQueueChain to handle serialization when acquiring a shared resource.
The SharedResourcesAcquirer now primarily holds a SerialTaskQueueChain and only holds mutexes for sharing between the Source and the DelayedReader. The SharedResourcesAcquirer are built to always contain at least one SerialTaskQueueChain since we no longer have a per module mutex and instead depends on having at least one queue.
This initial implementation uses the SerialTaskQueueChain in a synchronous fashion. Future changes will move to asynchronous usage.

ebrondol committed:
modify Producer to be compatible with Run2

ebrondol committed:
fix ordering in Phase2OTEndcapRing

ebrondol committed:
changing the name of the Phase2StipCPE

ianna committed:
New version of beampipe

ferencek committed:
removing JP, JBP, and cMVA taggers since JP calibrations can be provided for only one track selection at a time; adding c-taggers

ishvetso committed:
add puppi isolation

ishvetso committed:
line was removed accidentially

ishvetso committed:
add puppi isolation to PATPhotonProducer

alberto-sanchez committed:
adding the option of embed the user decay file for evtgen in the cfg itself

ishvetso committed:
fix vint32

ishvetso committed:
fix input

ishvetso committed:
fix patElectrons and patPhotons

Dr15Jones committed:
Handle case of very small numbers

We also need to ignore the case where the result returns a very small number, i.e. one with 'e-' in its value.

Dr15Jones committed:
Avoid race condition when exception happens in SerialTaskQueueChain

The decrementing of the ref count on the waiting task must be done after the exception is set in order to avoid the case where the wait_for_all is released and we return from pushAndWait before the value is assigned to the now gone away stack variable.

Dr15Jones committed:
Prefetching a legacy or one module no longer requires synchronizing

No longer do a SerialTaskQueueChain::pushAndWait when we are requested to run a legacy or one module as part of a data prefetch. Now the task can return immediately and the module will be run once the its task reaches the head of the queue. This required moving the use of the SerialTaskQueueChain to the task launched after the prefetch has completed.

prigoll committed:
MPS validation tool

Commit:
adjust indentation and empty line

gpetruc committed:
Switch on miniAOD for 2023 workflows that run RecoFullGlobal

gpetruc committed:
Add comments explainig the logic

ianna committed:
Adjust slope

ianna committed:
Format

Dr15Jones committed:
Removed mutexes from SharedResourcesAcquirer

The mutexes were only used for an InputSource. Now the InputSource gets an explicit mutex along with its SharedResourcesAcquirer.

alberto-sanchez committed:
example for a typical bph gen flow

Dr15Jones committed:
Emit post prefetch signal at correct point

Now the signal is emitted right after the post prefetch task has started rather than just right before we are going to run the module.

vbotta committed:
use das_client wrapper instead of the python das_client module

Dr15Jones committed:
Properly handle graphing stalled modules

Properly draw in the PDF file the case where a stalled module is running at the same time as a non-stalled module.

Dr15Jones committed:
Each Stream in graph is now a histogram of # running modules

The PDF now plots each stream as a histogram showing the number of concurrently running modules on that particular stream.

awhitbeck committed:
dropping HcalDbService, which is not used

rappoccio committed:
Fixed typeo in pruned mass histo

rappoccio committed:
Fixed typeo in soft drop mass histo

Commit:
Have some non-trivial parameters for SiPM hardcoded conditions

mariadalfonso committed:
remove never used double

ishvetso committed:
remove userFloat for puppi iso

Dr15Jones committed:
Turn on Services before emitting signal

Need to be sure all Services are active on a thread before emitting a signal since a Service listening to the signal may attempt to get another Service during that call.

Commit:
correct spelling mistake

ishvetso committed:
fix name

ferencek committed:
relaxing the default vertex track selection and overriding defaults in legacy SV TagInfos

ferencek committed:
propagating changes in the default b-tag track selection

Commit:
A trivial modification allowing indexing by subdetector

lgray committed:
code review

ishvetso committed:
temp? ...

ishvetso committed:
fix name

Commit:
bug fix in Phase2TrackerMonitorDigi.cc

cmsbuild commented:
A new Pull Request was created by  @ishvetso (Ivan Shvetsov) for CMSSW_8_1_X.

It involves the following packages:

DataFormats/PatCandidates
PhysicsTools/IsolationAlgos
PhysicsTools/PatAlgos
RecoEgamma/EgammaIsolationAlgos


@cmsbuild, @cvuosalo, @slava77, @montjj, @davidlange6 can you please review it and eventually sign? Thanks.
@rappoccio, @Sam-Harper, @JyothsnaKomaragiri, @rafaellopesdesa, @imarches, @ahinzmann, @acaudron, @mmarionncern, @TaiSakuma, @lgray, @jdolen, @nhanvtran, @gpetruc, @schoef, @ferencek, @mverzett, @mariadalfonso, @pvmulder, @cbernet this is something you requested to watch as well.
@slava77, @smuzaffar you are the release manager for this.

cms-bot commands are list here cms-sw/cmssw#13028

ishvetso commented on RecoEgamma/EgammaIsolationAlgos/plugins/isolation_cones/PhotonPFIsolationWithMapBasedVeto.cc:
```diff
@@ -132,9 +132,9 @@ bool PhotonPFIsolationWithMapBasedVeto::isInIsolationCone(const reco::CandidateP
-    inFootprint = isInFootprint(aspat_photonptr ->associatedPackedPFCandidates(),aspacked);
-    
-    //checking if the charged candidates come from the appropriate vertex
+    //inFootprint = isInFootprint(aspat_photonptr ->associatedPackedPFCandidates(),aspacked);
```
this is a line that gives trouble

slava77 replied:
Dereference of aspat_photonptr will work only if the input photon is a pat::Photon.
You configured the code to read reducedEgamma photons, which is made of reco::Photon.

This code will need to be updated to handle non-pat photons.

slava77 replied:
identical result can be figured out from 
https://github.com/cms-sw/cmssw/blob/CMSSW_8_1_0_pre10/PhysicsTools/PatAlgos/plugins/PATPhotonSlimmer.cc#L132

It would be more practical to save the reco-to-packed associations in 
ReducedEGProducer so that they are not re-derived again in the slimmers and  other code similar to this one

ishvetso replied:
do I understand correctly that in this case associatedPackedPFCandidates() function has to be introduced for reco::Photon?

slava77 replied:
effectively yes.
it will not be a member method of reco::Photon, just a function that can be computed for reco::Photon as an argument

ishvetso replied:
as far as I understand that means that this should be introduced in 
DataFormats/EgammaCandidates/interface/Photon.h
or you mean something else?

slava77 replied:
as I said,
it will not be a member method of reco::Photon, just a function that can be computed for reco::Photon as an argument

davidlange6 committed:
Merge pull request #15470 from mhl0116/from-CMSSW_8_0_14

Using an adjustable wire time window for CSC RecHit building

davidlange6 committed:
Merge pull request #15624 from Dr15Jones/fixStreamStallGrapher

Improve edmStreamStallGrapher

davidlange6 committed:
Merge pull request #15568 from kirschen/CMSSW_8_1_X_Fix_NANEta_CaloTowers

fix nan Calo-towers, when HO is deactivated, but has energy deposit (and ECAL/HCAL do not have any energy deposits)

slava77 commented on DataFormats/PatCandidates/src/classes_def_objects.xml:
```diff
@@ -16,7 +16,9 @@
-  <class name="pat::Electron"  ClassVersion="33">
+  <class name="pat::Electron"  ClassVersion="35">
+   <version ClassVersion="35" checksum="1857282274"/>
+   <version ClassVersion="34" checksum="3659401979"/>
```
please drop intermediate versions.
33 will need to go to only 34 for this PR alone.

ishvetso replied:
I cannot compile with 34 only. I get an error:
error: class 'pat::Electron' has a different checksum for ClassVersion 34. Increment ClassVersion to 35 and assign it to checksum 1857282274
Suggestion: You can run 'scram build updateclassversion' to generate src/DataFormats/PatCandidates/src/classes_def_objects.xml.generated with updated ClassVersion

slava77 replied:
remove current 34 and change remaining 35 to 34

slava77 commented on PhysicsTools/PatAlgos/plugins/PATElectronProducer.cc:
```diff
@@ -300,29 +312,32 @@ void PATElectronProducer::produce(edm::Event & iEvent, const edm::EventSetup & i
 	unsigned int idx = itElectron - electrons->begin();
-	if (Matched || MatchedToAmbiguousGsfTrack) continue;
+       auto elePtr = electrons -> ptrAt(idx);
+	     if (Matched || MatchedToAmbiguousGsfTrack) continue;
```
looks like indentation changed for no particular reason.
All edits not related to this PR purpose should be removed.
Given the list of not very informative commits in this PR overall, I think that for the final version they will need to be redone into just something finalized so that git history is not polluted.

ishvetso commented:
do you have any ideas about associatedPackedPFCandidates()? It is the main issue that is left now. If we cannot access it, that means that we have to run puppi, I think.

Dr15Jones committed:
Only use one thread for the test

The default for TBB is to use as many threads as cores. This is clearly unnecessary for this case so instead we just set it to one thread.
This test would infrequently fail with a strange message about missing file descriptor. I believe this was a byproduct of using so many threads.

wmtan committed:
Use unique_ptr, not auto_ptr

cmsbuild committed:
Merge pull request #15630 from Dr15Jones/stopFalsePositives

Only use one thread for the test

cmsbuild committed:
Merge pull request #15621 from vbotta/DASclientFix-81X-rebased

use das_client wrapper instead of the python das_client module

cmsbuild committed:
Merge pull request #15313 from Soureek89/topHLTOfflineDQM_July_27_2016_update_81X

Adding 2 new ElecMU paths to be monitored in topHLTOfflineDQM package for 81X

Dr15Jones committed:
Relax threshold for failure for no concurrent module test

Since the TriggerResults producer can always run concurrently, we need to relax the fraction of running 2 modules simultaneously from <0.01 to <0.02. This should avoid a false positive without missing a true failure.

Commit:
Modify TB code for CERN beam

ishvetso commented:
@slava77  do you have any feedback on this?

slava77 commented:
On 8/28/16 1:01 PM, Ivan Shvetsov wrote:
> @slava77 <https://github.com/slava77> do you have any feedback on this?

Nothing from a cursory look.
A more serious check will need time, which I may have in a week or so.
I suggested already to check with the egamma experts first.


> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/15628#issuecomment-242995889>, or
> mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEdcbt-rQLK0o23PPtBITG5RugTrZwDiks5qkekUgaJpZM4Jup8H>.
>

awhitbeck committed:
removing uneccessary include

wmtan committed:
Use unique_ptr, not auto_ptr

kfjack committed:
Add RAW+miniAOD data tier to Configuration/EventContent

jaylawhorn committed:
add more razor Hbb DQM modules

mariadalfonso committed:
fix the incremental version of the HBHEChannelInfo class

ebrondol committed:
moving strings to the constructor

cmsbuild committed:
Merge pull request #15611 from alberto-sanchez/embed-dec-81x

adding embedding of user.dec for evtgen in the cfg

davidlange6 committed:
Merge pull request #15596 from cms-tau-pog/CMSSW_8_1_X_tau-pog_quick-validation

Small cleanups and modifications of light weight tau s/w validation tools

davidlange6 committed:
Merge pull request #15437 from rappoccio/miniAOD81xJets

Jets for 81x MiniAOD

davidlange6 committed:
Merge pull request #15566 from suchandradutta/Phase2Digitizer_81X_160727

introduction of  Over-Threshold bit in outer tracker Digis

cmsbuild committed:
Merge pull request #15632 from bsunanda/Run2-TB07

Run2-TB07 Modify TB code for CERN beam

makortel committed:
Use HIP mitigation in 2015 and 2016B RelVals

lgray committed:
code review 2

Commit:
Give hardcoded #'s for TP parameters

ishvetso commented:
Let me post the error message, the error is coming from associatedPackedPFCandidates() as far as I understand:

A fatal system signal has occurred: segmentation violation
The following is the call stack containing the origin of the signal.

Thread 2 (Thread 0x7f15856bf700 (LWP 31956)):
#0  0x00007f159770c37d in waitpid () from /lib64/libpthread.so.0
#1  0x00007f1592453d47 in edm::service::cmssw_stacktrace_fork() () from /cvmfs/cms-ib.cern.ch/week1/slc6_amd64_gcc530/cms/cmssw/CMSSW_8_1_X_2016-08-25-2300/lib/slc6_amd64_gcc530/pluginFWCoreServicesPlugins.so
#2  0x00007f15924546d2 in edm::service::InitRootHandlers::stacktraceHelperThread() () from /cvmfs/cms-ib.cern.ch/week1/slc6_amd64_gcc530/cms/cmssw/CMSSW_8_1_X_2016-08-25-2300/lib/slc6_amd64_gcc530/pluginFWCoreServicesPlugins.so
#3  0x00007f1597c6da50 in std::(anonymous namespace)::execute_native_thread_routine (__p=<optimized out>) at ../../../../../libstdc++-v3/src/c++11/thread.cc:84
#4  0x00007f1597704aa1 in start_thread () from /lib64/libpthread.so.0
#5  0x00007f1597451aad in clone () from /lib64/libc.so.6
Thread 1 (Thread 0x7f1595b2abc0 (LWP 31913)):
#0  0x00007f1597448283 in poll () from /lib64/libc.so.6
#1  0x00007f1592454534 in full_read.constprop () from /cvmfs/cms-ib.cern.ch/week1/slc6_amd64_gcc530/cms/cmssw/CMSSW_8_1_X_2016-08-25-2300/lib/slc6_amd64_gcc530/pluginFWCoreServicesPlugins.so
#2  0x00007f159245479a in edm::service::InitRootHandlers::stacktraceFromThread() () from /cvmfs/cms-ib.cern.ch/week1/slc6_amd64_gcc530/cms/cmssw/CMSSW_8_1_X_2016-08-25-2300/lib/slc6_amd64_gcc530/pluginFWCoreServicesPlugins.so
#3  0x00007f159245508e in sig_dostack_then_abort () from /cvmfs/cms-ib.cern.ch/week1/slc6_amd64_gcc530/cms/cmssw/CMSSW_8_1_X_2016-08-25-2300/lib/slc6_amd64_gcc530/pluginFWCoreServicesPlugins.so
#4  <signal handler called>
#5  0x00007f1583c90ec9 in pat::Photon::associatedPackedPFCandidates() const () from /afs/cern.ch/work/i/ishvetso/EgammaWork/for_PR_electrons_v5/CMSSW_8_1_X_2016-08-25-2300/lib/slc6_amd64_gcc530/libDataFormatsPatCandidates.so
#6  0x00007f155df8f16a in PhotonPFIsolationWithMapBasedVeto::isInIsolationCone(edm::Ptr<reco::Candidate> const&, edm::Ptr<reco::Candidate> const&) const () from /afs/cern.ch/work/i/ishvetso/EgammaWork/for_PR_electrons_v5/CMSSW_8_1_X_2016-08-25-2300/lib/slc6_amd64_gcc530/pluginRecoEgammaEgammaIsolationAlgosPlugins_isolation_cones.so
#7  0x00007f155cf7ed27 in citk::PFIsolationSumProducerForPUPPI::produce(edm::Event&, edm::EventSetup const&) () from /afs/cern.ch/work/i/ishvetso/EgammaWork/for_PR_electrons_v5/CMSSW_8_1_X_2016-08-25-2300/lib/slc6_amd64_gcc530/pluginPhysicsToolsIsolationAlgos_plugins.so
#8  0x00007f1599dcb6b2 in edm::stream::EDProducerAdaptorBase::doEvent(edm::EventPrincipal const&, edm::EventSetup const&, edm::ActivityRegistry*, edm::ModuleCallingContext const*) () from /cvmfs/cms-ib.cern.ch/week1/slc6_amd64_gcc530/cms/cmssw/CMSSW_8_1_X_2016-08-25-2300/lib/slc6_amd64_gcc530/libFWCoreFramework.so
#9  0x00007f1599d282e2 in edm::WorkerT<edm::stream::EDProducerAdaptorBase>::implDo(edm::EventPrincipal const&, edm::EventSetup const&, edm::ModuleCallingContext const*) () from /cvmfs/cms-ib.cern.ch/week1/slc6_amd64_gcc530/cms/cmssw/CMSSW_8_1_X_2016-08-25-2300/lib/slc6_amd64_gcc530/libFWCoreFramework.so
#10 0x00007f1599ce5454 in bool edm::Worker::runModule<edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1> >(edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1>::MyPrincipal const&, edm::EventSetup const&, edm::StreamID, edm::ParentContext const&, edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1>::Context const*) () from /cvmfs/cms-ib.cern.ch/week1/slc6_amd64_gcc530/cms/cmssw/CMSSW_8_1_X_2016-08-25-2300/lib/slc6_amd64_gcc530/libFWCoreFramework.so
#11 0x00007f1599ce6216 in decltype ({parm#1}()) edm::convertException::wrap<void edm::Worker::runModuleAfterAsyncPrefetch<edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1> >(std::__exception_ptr::exception_ptr const*, edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1>::MyPrincipal const&, edm::EventSetup const&, edm::StreamID, edm::ParentContext const&, edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1>::Context const*)::{lambda()#1}>(void edm::Worker::runModuleAfterAsyncPrefetch<edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1> >(std::__exception_ptr::exception_ptr const*, edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1>::MyPrincipal const&, edm::EventSetup const&, edm::StreamID, edm::ParentContext const&, edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1>::Context const*)::{lambda()#1}) () from /cvmfs/cms-ib.cern.ch/week1/slc6_amd64_gcc530/cms/cmssw/CMSSW_8_1_X_2016-08-25-2300/lib/slc6_amd64_gcc530/libFWCoreFramework.so
#12 0x00007f1599ce635c in void edm::Worker::runModuleAfterAsyncPrefetch<edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1> >(std::__exception_ptr::exception_ptr const*, edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1>::MyPrincipal const&, edm::EventSetup const&, edm::StreamID, edm::ParentContext const&, edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1>::Context const*) () from /cvmfs/cms-ib.cern.ch/week1/slc6_amd64_gcc530/cms/cmssw/CMSSW_8_1_X_2016-08-25-2300/lib/slc6_amd64_gcc530/libFWCoreFramework.so
#13 0x00007f1599ce662d in edm::Worker::RunModuleTask<edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1> >::execute() () from /cvmfs/cms-ib.cern.ch/week1/slc6_amd64_gcc530/cms/cmssw/CMSSW_8_1_X_2016-08-25-2300/lib/slc6_amd64_gcc530/libFWCoreFramework.so
#14 0x00007f1598aad9b5 in tbb::internal::custom_scheduler<tbb::internal::IntelSchedulerTraits>::local_wait_for_all (this=0x7f1594b27000, parent=..., child=<optimized out>) at ../../src/tbb/custom_scheduler.h:463
#15 0x00007f1599ce5653 in decltype ({parm#1}()) edm::convertException::wrap<bool edm::Worker::doWork<edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1> >(edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1>::MyPrincipal const&, edm::EventSetup const&, edm::StreamID, edm::ParentContext const&, edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1>::Context const*)::{lambda()#2}>(bool edm::Worker::doWork<edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1> >(edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1>::MyPrincipal const&, edm::EventSetup const&, edm::StreamID, edm::ParentContext const&, edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1>::Context const*)::{lambda()#2}) () from /cvmfs/cms-ib.cern.ch/week1/slc6_amd64_gcc530/cms/cmssw/CMSSW_8_1_X_2016-08-25-2300/lib/slc6_amd64_gcc530/libFWCoreFramework.so
#16 0x00007f1599ce5a5d in bool edm::Worker::doWork<edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1> >(edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1>::MyPrincipal const&, edm::EventSetup const&, edm::StreamID, edm::ParentContext const&, edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1>::Context const*) () from /cvmfs/cms-ib.cern.ch/week1/slc6_amd64_gcc530/cms/cmssw/CMSSW_8_1_X_2016-08-25-2300/lib/slc6_amd64_gcc530/libFWCoreFramework.so
#17 0x00007f1599d025de in decltype ({parm#1}()) edm::convertException::wrap<void edm::Path::processOneOccurrence<edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1> >(edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1>::MyPrincipal const&, edm::EventSetup const&, edm::StreamID const&, edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1>::Context const*)::{lambda()#1}>(void edm::Path::processOneOccurrence<edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1> >(edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1>::MyPrincipal const&, edm::EventSetup const&, edm::StreamID const&, edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1>::Context const*)::{lambda()#1}) () from /cvmfs/cms-ib.cern.ch/week1/slc6_amd64_gcc530/cms/cmssw/CMSSW_8_1_X_2016-08-25-2300/lib/slc6_amd64_gcc530/libFWCoreFramework.so
#18 0x00007f1599d02892 in void edm::Path::processOneOccurrence<edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1> >(edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1>::MyPrincipal const&, edm::EventSetup const&, edm::StreamID const&, edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1>::Context const*) () from /cvmfs/cms-ib.cern.ch/week1/slc6_amd64_gcc530/cms/cmssw/CMSSW_8_1_X_2016-08-25-2300/lib/slc6_amd64_gcc530/libFWCoreFramework.so
#19 0x00007f1599d02dcc in void edm::StreamSchedule::processOneEvent<edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1> >(edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1>::MyPrincipal&, edm::EventSetup const&, bool)::{lambda()#1}::operator()() const () from /cvmfs/cms-ib.cern.ch/week1/slc6_amd64_gcc530/cms/cmssw/CMSSW_8_1_X_2016-08-25-2300/lib/slc6_amd64_gcc530/libFWCoreFramework.so
#20 0x00007f1599d02fab in decltype ({parm#1}()) edm::convertException::wrap<void edm::StreamSchedule::processOneEvent<edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1> >(edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1>::MyPrincipal&, edm::EventSetup const&, bool)::{lambda()#1}>(void edm::StreamSchedule::processOneEvent<edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1> >(edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1>::MyPrincipal&, edm::EventSetup const&, bool)::{lambda()#1}) () from /cvmfs/cms-ib.cern.ch/week1/slc6_amd64_gcc530/cms/cmssw/CMSSW_8_1_X_2016-08-25-2300/lib/slc6_amd64_gcc530/libFWCoreFramework.so
#21 0x00007f1599d0321f in void edm::StreamSchedule::processOneEvent<edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1> >(edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1>::MyPrincipal&, edm::EventSetup const&, bool) () from /cvmfs/cms-ib.cern.ch/week1/slc6_amd64_gcc530/cms/cmssw/CMSSW_8_1_X_2016-08-25-2300/lib/slc6_amd64_gcc530/libFWCoreFramework.so
#22 0x00007f1599d80182 in edm::EventProcessor::processEvent(unsigned int) () from /cvmfs/cms-ib.cern.ch/week1/slc6_amd64_gcc530/cms/cmssw/CMSSW_8_1_X_2016-08-25-2300/lib/slc6_amd64_gcc530/libFWCoreFramework.so
#23 0x00007f1599d809fa in edm::EventProcessor::processEventsForStreamAsync(unsigned int, std::atomic<bool>*) () from /cvmfs/cms-ib.cern.ch/week1/slc6_amd64_gcc530/cms/cmssw/CMSSW_8_1_X_2016-08-25-2300/lib/slc6_amd64_gcc530/libFWCoreFramework.so
#24 0x00007f1599d871b4 in edm::StreamProcessingTask::execute() () from /cvmfs/cms-ib.cern.ch/week1/slc6_amd64_gcc530/cms/cmssw/CMSSW_8_1_X_2016-08-25-2300/lib/slc6_amd64_gcc530/libFWCoreFramework.so
#25 0x00007f1598aad9b5 in tbb::internal::custom_scheduler<tbb::internal::IntelSchedulerTraits>::local_wait_for_all (this=0x7f1594b27000, parent=..., child=<optimized out>) at ../../src/tbb/custom_scheduler.h:463
#26 0x00007f1599d8070e in edm::EventProcessor::readAndProcessEvent() () from /cvmfs/cms-ib.cern.ch/week1/slc6_amd64_gcc530/cms/cmssw/CMSSW_8_1_X_2016-08-25-2300/lib/slc6_amd64_gcc530/libFWCoreFramework.so
#27 0x00007f1599ced3da in statemachine::HandleEvent::readAndProcessEvent() () from /cvmfs/cms-ib.cern.ch/week1/slc6_amd64_gcc530/cms/cmssw/CMSSW_8_1_X_2016-08-25-2300/lib/slc6_amd64_gcc530/libFWCoreFramework.so
#28 0x00007f1599cee968 in statemachine::HandleEvent::HandleEvent(boost::statechart::state<statemachine::HandleEvent, statemachine::HandleLumis, boost::mpl::list<mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na>, (boost::statechart::history_mode)0>::my_context) () from /cvmfs/cms-ib.cern.ch/week1/slc6_amd64_gcc530/cms/cmssw/CMSSW_8_1_X_2016-08-25-2300/lib/slc6_amd64_gcc530/libFWCoreFramework.so
#29 0x00007f1599cf66ae in boost::statechart::state<statemachine::HandleEvent, statemachine::HandleLumis, boost::mpl::list<mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na>, (boost::statechart::history_mode)0>::shallow_construct(boost::intrusive_ptr<statemachine::HandleLumis> const&, boost::statechart::state_machine<statemachine::Machine, statemachine::Starting, std::allocator<void>, boost::statechart::null_exception_translator>&) () from /cvmfs/cms-ib.cern.ch/week1/slc6_amd64_gcc530/cms/cmssw/CMSSW_8_1_X_2016-08-25-2300/lib/slc6_amd64_gcc530/libFWCoreFramework.so
#30 0x00007f1599cf687a in boost::statechart::simple_state<statemachine::FirstLumi, statemachine::HandleLumis, boost::mpl::list<mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na>, (boost::statechart::history_mode)0>::react_impl(boost::statechart::event_base const&, void const*) () from /cvmfs/cms-ib.cern.ch/week1/slc6_amd64_gcc530/cms/cmssw/CMSSW_8_1_X_2016-08-25-2300/lib/slc6_amd64_gcc530/libFWCoreFramework.so
#31 0x00007f1599d86c68 in boost::statechart::state_machine<statemachine::Machine, statemachine::Starting, std::allocator<void>, boost::statechart::null_exception_translator>::process_event(boost::statechart::event_base const&) () from /cvmfs/cms-ib.cern.ch/week1/slc6_amd64_gcc530/cms/cmssw/CMSSW_8_1_X_2016-08-25-2300/lib/slc6_amd64_gcc530/libFWCoreFramework.so
#32 0x00007f1599d7a05d in edm::EventProcessor::runToCompletion() () from /cvmfs/cms-ib.cern.ch/week1/slc6_amd64_gcc530/cms/cmssw/CMSSW_8_1_X_2016-08-25-2300/lib/slc6_amd64_gcc530/libFWCoreFramework.so
#33 0x000000000040c844 in main::{lambda()#1}::operator()() const ()
#34 0x000000000040acc4 in main ()

Current Modules:

Module: CITKPFIsolationSumProducerForPUPPI (crashed)

A fatal system signal has occurred: segmentation violation

Commit:
Update the python script as well

cmsbuild committed:
Merge pull request #15636 from jaylawhorn/razor-trigger-update-08-29-16-81X

add more razor Hbb DQM modules

davidlange6 committed:
Merge pull request #15605 from gpetruc/tkLayersInMiniAOD-81X

Save pixel and strip layers in miniAOD

cmsbuild committed:
Merge pull request #15575 from uhussain/fix_staticAnaWarns

Fixed Logic Errors in CalibrationXML.cc, SiStripElectronAlgo.cc & InOutConversionSeedFinder.cc

cmsbuild committed:
Merge pull request #15520 from awhitbeck/HCalUTCApacker

HCal uTCA packer

cmsbuild committed:
Merge pull request #15608 from ebrondol/fixPhase2OTEndcapRing_81X

Fix ordering in Phase2OTEndcapRing

Commit:
Update the customization script as well

cmsbuild committed:
Merge pull request #15563 from makortel/fixPhase1MissCalibrate

Set siPixelClusters.MissCalibrate=True for phase1

boudoul committed:
set Pilot Blade to false by default

ianna committed:
* Cleanup remaining 2023 scenarios

smuzaffar committed:
fixes need to compile against latest root

smuzaffar committed:
fix needed due to cleanup in root6

prigoll committed:
fixed bugs mentioned in #15642

cmsbuild committed:
Merge pull request #15606 from kpedro88/Phase1-DbHardcode2

Automatic emap in HCAL hardcode conditions

cmsbuild committed:
Merge pull request #15550 from Michael-Krohn/AdditionalMaterials

Added the radiation length and hadronic interaction length of unclassified materials

davidlange6 committed:
Merge pull request #15421 from cms-btv-pog/bTagHIPMitigation-PR_from-CMSSW_8_1_X_2016-08-10-1100

btag HIP mitigation + ctag bugfix

cmsbuild committed:
Merge pull request #15499 from mariadalfonso/M2

HE- energy reconstruction - phase1

mmusich committed:
rolling back bad updates introduced for Tranche-IV MC

davidlange6 committed:
Merge pull request #15656 from ianna/2023-scenarios-cleanup

Usage of Obsolete 2023 Scenarios Cleanup

cmsbuild committed:
Merge pull request #15646 from bsunanda/Run2-alca61

bsunanda:Run2-alca61 Give hardcoded #'s for TP parameters

ferencek committed:
updated if statements with correct python logic

deguio committed:
fix and include the DigiPhase1Task in the phase1 customization function

Dr15Jones committed:
Ignore Finish Init entry when making PDF

Dr15Jones committed:
Added isUsingModifier to Process

This replaces the deprecated call to Modifier.isChosen. This interface can be supported by future improvements to the code (where Modifier.isChosen can not be).

cmsbuild committed:
Merge pull request #15654 from boudoul/Pilot

set Pilot Blade to false by default

davidlange6 committed:
Merge pull request #15655 from smuzaffar/fixes-for-root6-master

Fixes for root6 master

davidlange6 committed:
Merge pull request #15657 from smuzaffar/root6-fixes-analysis

fix needed due to cleanup in root6

cmsbuild committed:
Merge pull request #15642 from prigoll/mpsvalidate

Add MPS validation tool

davidlange6 committed:
Merge pull request #15627 from igv4321/HcalDetIdTransform-update

A trivial modification allowing item indexing by subdetector

wmtan committed:
Use unique_ptr, not auto_ptr

Commit:
Create a context around TFile::Open() so the context is unregistered in the same threaded it was registered

cmsbuild committed:
Merge pull request #15670 from dan131riley/unregister-tfile-context

Create a context around TFile::Open() so the context is unregistered …

cmsbuild committed:
Merge pull request #15667 from Dr15Jones/runModuleAsync_nonDevel

Use SerialTaskQueues to protect shared resources (8_1)

cmsbuild committed:
Merge pull request #15664 from Dr15Jones/fixStreamGrapherPDF

Ignore Finish Init entry when making PDF

davidlange6 committed:
Merge pull request #15598 from lgray/hgcal_global_reco

HGCal Global Reco (using "Sim Particle Flow" in endcaps)

cmsbuild committed:
Merge pull request #15649 from smorovic/parse-json-and-selective-stop-81X

New stop-run control and more debugging info on JSON parse error in input source (DAQ) - 81X

davidlange6 committed:
Merge pull request #15665 from Dr15Jones/processIsUsingModifier

Added isUsingModifier to Process

davidlange6 committed:
Merge pull request #15658 from mmusich/81X-revert-badUpdatesTrancheIV

rolling back bad updates introduced for Tranche-IV MC

davidlange6 committed:
Merge pull request #15626 from bsunanda/Run2-alca60

bsunanda:Run2-alca60 Have some non-trivial parameters for SiPM hardcoded conditions

thuer committed:
Fix mem leak in rivet analyzer

ebrondol committed:
moving getParameter of CPE strings in the constructor

Commit:
2016 pPb HLT DQM for HIN flow group

makortel committed:
Enable HIP mitigation in 2015 only to data (keep disabled for MC)

thuer committed:
Changed pointer to smart pointer and remove delete statement

ianna committed:
Merged postls2-beampipe-v5.4.0 from repository ianna

ianna committed:
* Unit test with 2023 scenarios failure fix

ferencek committed:
with tracker layers info in MiniAOD, no longer need to relax cut on the number of layers

kpedro88 committed:
fix memory leak in HcalDbService

gpetruc committed:
photonIDValueMapProducer: don't use ES in phase2

gpetruc committed:
EcalDeadCellTriggerPrimitiveFilter: don't use EE in Phase2/HGCal era (c++ code change also needed to avoid crash in beginRun)

gpetruc committed:
Disable MET filters that don't work in current Phase2/HGCal era (only when that era is selected, of course)

gpetruc committed:
Take out MET MiniAOD DQM in Phase2/HGCal era, for which hcalnoise is missing

Commit:
Modify TDC mask from 32 to 64 bit

Commit:
Fixing M2 call arguments in SimpleHBHEPhase1Algo

cmsbuild committed:
Merge pull request #15686 from kpedro88/FixHcalDbLeak

fix memory leak in HcalDbService

cmsbuild committed:
Merge pull request #15668 from wmtan/UseUniquePtrInMiscReco

Use unique_ptr, not auto_ptr in some Reco* packages

cmsbuild committed:
Merge pull request #15635 from wmtan/UseUniquePtrInRecoLocalTracker

Use unique_ptr, not auto_ptr in RecoLocalTracker

cmsbuild committed:
Merge pull request #15612 from ebrondol/phase2trackingWithCPE_81X

Importing Phase2StripCPE from external package

davidlange6 committed:
Merge pull request #15631 from wmtan/UseUniquePtrInEventFilter

Use unique_ptr, not auto_ptr in EventFilter

cmsbuild committed:
Merge pull request #15663 from deguio/hcalPhase1-monitoring

add DigiPhase1Task

davidlange6 committed:
Merge pull request #15681 from makortel/enablePhase1Raw2Digi_v3

Enable pixel digi2raw and raw2digi modules in 2017 workflows

davidlange6 committed:
Merge pull request #15590 from makortel/VinInn_NoVFP30Mitigation

No vfp30 mitigation in Tracking (including customization with era)

davidlange6 committed:
Merge pull request #15659 from cms-btv-pog/jetToolsFix-PR_from_CMSSW_8_1_X_b7fcf95e049f3fb64fef18971942f63c7d6ff685

Fixed if statements in jetTools.py

mmusich committed:
update MC GTs with conditions validated for MC Tranche-IV and prepare for data reprocessing

cmsbuild committed:
Merge pull request #15673 from thuer/Rivet_memleak_fix

Fix mem leak in rivet analyzer

ianna committed:
* Add description upto the end of the hall
* Remove Castor and Totem
* Modify Tracker envelopes

mariadalfonso committed:
fix uninitialized vectors

fwyzard committed:
HLT tools: move the default to ConfDB v2

ianna committed:
* Move modified Pixel envelope in a PostLS2 dir

ianna committed:
* Restore PhaseI Pixel envelope

ianna committed:
* All Post LS2 modifications in corresponding PostLS2 dirs

ianna committed:
Restore old Phase II beampipe

smuzaffar committed:
Merge pull request #15680 from ianna/2023-scenarios-unit-test-fix

2023 Scenarios Unit Test Fix

ianna committed:
Reduce support ring to fit Tracer envelope.

Commit:
Add -fno-crossjumping to black list

Clang does not support -fno-crossjumping flag thus adding to the
blacklist.

Signed-off-by: David Abdurachmanov <David.Abdurachmanov@cern.ch>

davidlt committed:
Merge pull request #15699 from davidlt/add-fno-cross-jump-blacklist

Add -fno-crossjumping to black list

davidlange6 committed:
Merge pull request #15633 from kfjack/add-raw-miniaod-tier

Add RAW+miniAOD data tier to Configuration/EventContent

davidlange6 committed:
Merge pull request #15615 from gpetruc/miniAOD_phase2

MiniAOD in phase2

cmsbuild committed:
Merge pull request #15697 from fwyzard/set_confdbv2_by_default_81x

HLT tools: move the default to ConfDB v2

cmsbuild committed:
Merge pull request #15694 from mmusich/81X_GTprepareForMCTrancheIVandDataReprocessing

update MC GTs with conditions validated for MC Tranche-IV and prepare for 2016 data reprocessing

cmsbuild committed:
Merge pull request #15693 from ianna/postls2-beampipe-v5.4.0

Post LS2 Beam-pipe v5.4.0

wmtan committed:
Avoid unnecessary function calls by interpreter

davidlange6 committed:
Merge pull request #15684 from cms-btv-pog/IVFtrackMinLayersForMiniAOD-PR_from-CMSSW_8_1_X_2016-08-31-1500

Updated IVF trackMinLayers cut in jetTools.py for MiniAOD

cmsbuild committed:
Merge pull request #15688 from bsunanda/Run2-alca62

bsunanda:Run2-alca62 Modify TDC mask in HcalTPParameters from 32 bit to 64 bit uint

cmsbuild committed:
Merge pull request #14422 from JetMETdqmval/DQM_81X_test_PFCandidate_afterRebase

Dqm 81 x test pf candidate after rebase

cmsbuild committed:
Merge pull request #15678 from zc11/HLTDQM2016pPbFlow81X

2016 pPb HLT DQM for HIN flow group

ishvetso commented:
I think the reason for the crash is that pat object is not being created properly from "reducedEgamma","reducedGedPhotons". This is the way it is done:
pat::patPhotonPtr aspat_photonptr(photon);

where photon is reco::Candidate pointer. Do you have any ideas why this happens?

davidlange6 committed:
Revert "Dqm 81 x test pf candidate after rebase"

davidlange6 committed:
Merge pull request #15690 from igv4321/SimpleHBHEPhase1Algo-m2-fix

Fixing Method 2 use in SimpleHBHEPhase1Algo

davidlange6 committed:
Merge pull request #15695 from mariadalfonso/FixCrash

HBHE: Fix uninitialized vectors (to avoid crash)

davidlange6 committed:
Merge pull request #15718 from cms-sw/revert-14422-DQM_81X_test_PFCandidate_afterRebase

Revert "Dqm 81 x test pf candidate after rebase"

davidlange6 committed:
Merge pull request #15595 from cms-tau-pog/CMSSW_8_1_X_tau-pog_new-bend-config

Modifications of bend corrections to improve tau reconstruction performance

davidlange6 committed:
Merge pull request #15700 from wmtan/AvoidUnneededClingInterpreterCalls

Bug Fix! Avoid unnecessary function calls by CLING interpreter in cutParser

slava77 commented on DataFormats/PatCandidates/interface/Electron.h:
```diff
@@ -160,6 +160,17 @@ namespace pat {
 
+      float puppiChargedHardonsIso() const {return puppiChargedHardonsIso_; };
+      float puppiNeutralHardonsIso() const {return puppiNeutralHardonsIso_; };
+      float puppiPhotonsIso() const {return puppiPhotonsIso_; };
```
trailing semicolons (after }) are not necessary

slava77 commented on DataFormats/PatCandidates/interface/Photon.h:
```diff
@@ -169,6 +169,11 @@ namespace pat {
+
+      float puppiChargedHardonsIso() const {return puppiChargedHardonsIso_; };
+      float puppiNeutralHardonsIso() const {return puppiNeutralHardonsIso_; };
+      float puppiPhotonsIso() const {return puppiPhotonsIso_; };
```
trailing semicolons (after }) are not necessary

slava77 commented on PhysicsTools/IsolationAlgos/plugins/CITKPFIsolationSumProducerForPUPPI.cc:
```diff
@@ -153,7 +156,8 @@ namespace citk {
           double puppiWeight = 0.;
-    	    if (!useValueMapForPUPPI) puppiWeight = aspackedCandidate -> puppiWeight(); // if miniAOD, take puppiWeight directly from the object
+    	    if (!useValueMapForPUPPI && !usePUPPINoLepton) puppiWeight = aspackedCandidate -> puppiWeight(); // if miniAOD, take puppiWeight directly from the object
+          else if (!useValueMapForPUPPI && usePUPPINoLepton) puppiWeight = aspackedCandidate -> puppiWeightNoLep(); // if miniAOD, take puppiWeightNoLep directly from the object
```
looks like the two lines above have inconsistent indentation.
Please adjust

slava77 commented on PhysicsTools/PatAlgos/plugins/PATElectronSlimmer.cc:
```diff
@@ -20,6 +20,7 @@
 #include "RecoEcal/EgammaCoreTools/interface/EcalClusterLazyTools.h"
 #include "FWCore/Utilities/interface/isFinite.h"
+#include "DataFormats/Common/interface/ValueMap.h"
 
```
there are no useful changes in this file anymore.
Better remove all edits, better yet ```git rebase -i ``` and remove commits that modified this file.

slava77 commented on PhysicsTools/PatAlgos/python/slimming/egmIsolationsPUPPI_cfi.py:
```diff
@@ -1,10 +1,22 @@
+
+	process.egmElectronPUPPINoLeptonsIsolation = _egmElectronIsolationMiniAODPUPPINoLeptons.clone()
+	process.egmElectronPUPPINoLeptonsIsolation.srcToIsolate = cms.InputTag("reducedEgamma","reducedGedGsfElectrons")
+	process.egmElectronPUPPINoLeptonsIsolation.srcForIsolationCone = cms.InputTag("packedPFCandidates")
```
newline is missing

ishvetso replied:
fixed

slava77 commented on PhysicsTools/PatAlgos/python/slimming/egmIsolationsPUPPI_cfi.py:
```diff
@@ -1,10 +1,22 @@
-	process.egmPhotonPUPPIIsolationForPhotons = _egmPhotonPUPPIIsolationForPhotons.clone()
-	process.egmPhotonPUPPIIsolationForPhotons.srcToIsolate = cms.InputTag("selectedPatPhotons")
-	process.egmPhotonPUPPIIsolationForPhotons.srcForIsolationCone = cms.InputTag("packedPFCandidates")
+	process.load('RecoEgamma.EgammaIsolationAlgos.egmElectronIsolationPUPPI_cff')
```
process.load calls appear unnecessary given the explicit imports made above (lines 3-5).

ishvetso committed:
Merged PUPPI_isolation_electrons_miniAOD_rebased from repository ishvetso

ishvetso committed:
Update CITKPFIsolationSumProducerForPUPPI.cc

fix indentation

cmsbuild commented:
Pull request #15628 was updated. @cmsbuild, @cvuosalo, @slava77, @monttj, @davidlange6 can you please check and sign again.

ishvetso committed:
remove process loads

cmsbuild commented:
Pull request #15628 was updated. @cmsbuild, @cvuosalo, @slava77, @monttj, @davidlange6 can you please check and sign again.

ishvetso committed:
Merge branch 'PUPPI_isolation_electrons_miniAOD_rebased' of https://github.com/ishvetso/cmssw into from-CMSSW_8_1_X_2016-09-04-0000

ishvetso committed:
update indentation

cmsbuild commented:
Pull request #15628 was updated. @ghellwig, @covarell, @perrozzi, @ianna, @cerminar, @fabozzi, @Martin-Grunewald, @mommsen, @monttj, @govoni, @perrotta, @civanch, @vanbesien, @silviodonato, @franzoni, @cmsbuild, @fwyzard, @davidlange6, @smuzaffar, @Dr15Jones, @cvuosalo, @mdhildreth, @slava77, @ggovi, @mmusich, @mulhearn, @rekovic, @emeschi, @thuer, @dmitrijus, @alja, @srimanob, @hengne can you please check and sign again.

slava77 commented:
-1
needs a rebase

ishvetso commented:
please refer to #15722
